eth/handler.go

```diff
@@ -173,7 +173,7 @@ func NewProtocolManager(config *params.ChainConfig, fastSync bool, networkId int
 		return blockchain.CurrentBlock().NumberU64()
 	}
 	inserter := func(blocks types.Blocks) (int, error) {
-		manager.setSynced() // Mark initial sync done on any fetcher import
+		atomic.StoreUint32(&manager.synced, 1) // Mark initial sync done on any fetcher import
```

Comment:
No, synced is being read and written in different goroutines, see:
https://github.com/ethereum/go-ethereum/blob/master/eth/handler.go#L663
This is how it worked before I added setSynced (I restored it in its original state).